modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/NameValueResolver.java

```diff
@@ -0,0 +1,115 @@
+package com.enonic.xp.core.impl.content;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+abstract class NameValueResolver
+{
+    private static final String NAME_SEPARATOR = "-";
+
+    public static String name( final String existingName )
+    {
+        return doResolve( existingName, NAME_SEPARATOR );
+    }
+
+    private static String doResolve( final String existingName, final String separator )
+    {
+        final NewName newName = NewName.from( existingName, separator );
+
+        return newName.getNextValue();
+    }
+
+    private static final class NewName
+    {
+        Integer copyCounter;
+
+        String baseName;
+
+        private String separator;
+
+        private NewName()
```

Comment:
Codacy found an issue: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/gh/enonic/xp/pullRequest?prid=6409742)